src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -605,6 +606,39 @@ public Node visitCastExpr(JsoniqParser.CastExprContext ctx) {
         return new CastExpression(mainExpression, sequenceType, createMetadataFromContext(ctx));
     }
 
+    @Override
+    public Node visitArrowExpr(JsoniqParser.ArrowExprContext ctx) {
+        Expression mainExpression = (Expression) this.visitUnaryExpr(ctx.main_expr);
+
+        for (int i = 0; i < ctx.function_call_expr.size(); ++i) {
+            FunctionCallContext functionCallContext = ctx.function_call_expr.get(i);
+            String name;
+            if (functionCallContext.fn_name != null) {
+                name = functionCallContext.fn_name.getText();
+            } else {
+                name = functionCallContext.kw.getText();
+            }
+            if (functionCallContext.ns != null) {
+                name = name + ":" + functionCallContext.ns.getText();
+            }
+            List<Expression> arguments = new ArrayList<>();
+            arguments.add(mainExpression);
+            if (functionCallContext.argumentList().args != null) {
+                for (JsoniqParser.ArgumentContext arg : functionCallContext.argumentList().args) {
+                    Expression currentArg = (Expression) this.visitArgument(arg);
+                    arguments.add(currentArg);
+                }
+            }
```

Comment:
Similarly, the contents of this if block is identical to the getArgumentsFromArgumentListContext() method. Can we call that method and join the returned list with the arguments list here?